### ISSUE TITLE
TestNG's threadPoolFactoryClass parameter broken for TestNG >= 7.10

### ISSUE THREAD

**Issue by clayburn** [gradle/gradle#28955]
### Current Behavior

Using TestNG 10.7.1, the latest version, a `org.gradle.api.InvalidUserDataException: The version of TestNG used does not support setting thread pool factory class` exception is thrown when setting `threadPoolFactoryClass`.

### Expected Behavior

Using TestNG 10.7.1, Gradle should allow configuration of the `threadPoolFactoryClass` option.

### Context (optional)

I think the issue is that Gradle is reflectively checking for the presence of `setExecutorFactoryClass` on `org.testng.testNG`. However, it looks like this changed to `setExecutorServiceFactory` in [this PR](https://github.com/testng-team/testng/pull/3075/files#diff-fd7fcf3b0ee340e791a1bdb372f4eeef4986d4daec08691434053a6079c8d092L846) which went into 7.10.

### Steps to Reproduce

github.com/clayburn/testng-issue

### Gradle version

8.7

### Build scan URL (optional)

https://ge.solutions-team.gradle.com/s/4oz3keudfzdie

**Comment by ov7a** [comment]
The issue is in the backlog of the relevant team and is prioritized by them.

**Comment by big-guy** [comment]
Yeah, we're checking for `setExecutorFactoryClass(String clazzName)` and if it doesn't exist, we don't think TestNG supports the option.

We would need to start looking for both with the new method `setExecutorServiceFactory(IExecutorServiceFactory factory)`.  I'm not sure how we get an instance of that factory. 

PRs welcome.
